sqlmesh/core/engine_adapter/snowflake.py

```diff
@@ -72,28 +75,28 @@ def session(self, properties: SessionProperties) -> t.Iterator[None]:
         yield
         self.execute(f"USE WAREHOUSE {current_warehouse_sql}")
 
+    @property
+    def snowpark(self) -> t.Optional[SnowparkSession]:
+        if snowpark:
+            return snowpark.Session.builder.configs(
+                {"connection": self._connection_pool.get()}
+            ).getOrCreate()
+        return None
+
     def _df_to_source_queries(
         self,
         df: DF,
         columns_to_types: t.Dict[str, exp.DataType],
         batch_size: int,
         target_table: TableName,
     ) -> t.List[SourceQuery]:
-        assert isinstance(df, pd.DataFrame)
         temp_table = self._get_temp_table(target_table or "pandas")
 
         def query_factory() -> Query:
-            from snowflake.connector.pandas_tools import write_pandas
-
-            if not self.table_exists(temp_table):
-                # Workaround for https://github.com/snowflakedb/snowflake-connector-python/issues/1034
-                #
-                # The above issue has already been fixed upstream, but we keep the following
-                # line anyway in order to support a wider range of Snowflake versions.
```

Comment:
If you are going to drop support for older versions, we should update the min version on Snowflake then right to reflect that?

Reply:
this bug only affects snowflake users on old versions who happen to also use pandas. i could add a min but didn't think it was that necessary

Reply:
after reading the dbt issues related to this, i think i'll keep the workaround for now